File: WriterLab-v1/fastapi/backend/app/api/projects.py
from uuid import UUID

from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session

from app.db.session import get_db
from app.models.project import Project
from app.repositories.project_repository import (
    delete_project as delete_project_query,
    list_projects as list_projects_query,
)
from app.schemas.project import ProjectCreate, ProjectDeleteResponse, ProjectResponse

router = APIRouter(prefix="/api/projects", tags=["projects"])


@router.post("", response_model=ProjectResponse)
def create_project(payload: ProjectCreate, db: Session = Depends(get_db)):
    project = Project(
        name=payload.name,
        description=payload.description,
        genre=payload.genre,
        default_language=payload.default_language,
    )
    db.add(project)
    db.commit()
    db.refresh(project)
    return project


@router.get("", response_model=list[ProjectResponse])
def list_projects(db: Session = Depends(get_db)):
    return list_projects_query(db)


@router.delete("/{project_id}", response_model=ProjectDeleteResponse)
def delete_project(project_id: UUID, db: Session = Depends(get_db)):
    if not delete_project_query(db, project_id):
        raise HTTPException(status_code=404, detail="Project not found")
    return {"deleted": True, "project_id": project_id}

File: WriterLab-v1/Next.js/frontend/lib/api/projects.ts
import { apiDelete, apiGet, apiPost } from "@/lib/api/client";

export type ProjectCreatePayload = {
  name: string;
  description?: string;
  genre?: string;
  default_language?: string;
};

export type ProjectResponse = {
  id: string;
  name: string;
  description?: string | null;
  genre?: string | null;
  default_language: string;
  created_at: string;
  updated_at: string;
};

export type ProjectDeleteResponse = {
  deleted: boolean;
  project_id: string;
};

export function fetchProjects<T>() {
  return apiGet<T>("/api/projects", "读取项目失败");
}

export function createProject<T>(body: ProjectCreatePayload) {
  return apiPost<T>("/api/projects", body, "创建项目失败");
}

export function deleteProject<T>(projectId: string) {
  return apiDelete<T>(`/api/projects/${projectId}`, "删除项目失败");
}

export function fetchBooksByProject<T>(projectId: string) {
  return apiGet<T>(`/api/books?project_id=${projectId}`, "读取书籍失败");
}

export function fetchChaptersByBook<T>(bookId: string) {
  return apiGet<T>(`/api/chapters?book_id=${bookId}`, "读取章节失败");
}
